src/rcl_lifecycle_bindings.cpp

```diff
@@ -0,0 +1,363 @@
+             wrapState(transition.start));
+    Nan::Set(jsTransitionDesc, Nan::New("goal_state").ToLocalChecked(),
+             wrapState(transition.goal));
+
+    Nan::Set(jsTransitions, i, jsTransitionDesc);
+  }
+
+  info.GetReturnValue().Set(jsTransitions);
+}
+
+// return the transitions available from the current state
+NAN_METHOD(GetAvailableLifecycleTransitions) {
+  v8::Local<v8::Context> currentContent = Nan::GetCurrentContext();
+  RclHandle* state_machine_handle = RclHandle::Unwrap<RclHandle>(
+      Nan::To<v8::Object>(info[0]).ToLocalChecked());
+  rcl_lifecycle_state_machine_t* state_machine =
+      reinterpret_cast<rcl_lifecycle_state_machine_t*>(
+          state_machine_handle->ptr());
+
+  v8::Local<v8::Array> jsTransitions = Nan::New<v8::Array>();
+
+  for (uint8_t i = 0; i < state_machine->current_state->valid_transition_size;
+       ++i) {
+    auto transition = state_machine->current_state->valid_transitions[i];
+    v8::Local<v8::Object> jsTransitionDesc = Nan::New<v8::Object>();
+    Nan::Set(jsTransitionDesc, Nan::New("transition").ToLocalChecked(),
+             wrapTransition(&transition));
+    Nan::Set(jsTransitionDesc, Nan::New("start_state").ToLocalChecked(),
+             wrapState(transition.start));
+    Nan::Set(jsTransitionDesc, Nan::New("goal_state").ToLocalChecked(),
+             wrapState(transition.goal));
+
+    Nan::Set(jsTransitions, i, jsTransitionDesc);
+  }
+
+  info.GetReturnValue().Set(jsTransitions);
+}
+
+NAN_METHOD(GetLifecycleSrvNameAndHandle) {
+  v8::Local<v8::Context> currentContent = Nan::GetCurrentContext();
+  RclHandle* node_handle = RclHandle::Unwrap<RclHandle>(
+      Nan::To<v8::Object>(info[0]).ToLocalChecked());
+  rcl_node_t* node = reinterpret_cast<rcl_node_t*>(node_handle->ptr());
+
+  RclHandle* state_machine_handle = RclHandle::Unwrap<RclHandle>(
+      Nan::To<v8::Object>(info[1]).ToLocalChecked());
+  rcl_lifecycle_state_machine_t* state_machine =
+      reinterpret_cast<rcl_lifecycle_state_machine_t*>(
+          state_machine_handle->ptr());
+
+  std::string lifecycle_srv_field_name(*Nan::Utf8String(info[2]));
+
+  std::string service_name;
+  rcl_service_t* service;
+  if (lifecycle_srv_field_name.compare("srv_get_state") == 0) {
+    service_name = "~/get_state";
+    service = &state_machine->com_interface.srv_get_state;
+  } else if (lifecycle_srv_field_name.compare("srv_get_available_states") ==
+             0) {
+    service_name = "~/get_available_states";
+    service = &state_machine->com_interface.srv_get_available_states;
+  } else if (lifecycle_srv_field_name.compare(
+                 "srv_get_available_transitions") == 0) {
+    service_name = "~/get_available_transitions";
+    service = &state_machine->com_interface.srv_get_available_transitions;
+  } else if (lifecycle_srv_field_name.compare("srv_change_state") == 0) {
+    service_name = "~/change_state";
+    service = &state_machine->com_interface.srv_change_state;
+  }
```

Comment:
missing find `else` so we may end up creating a `RclHandle` to a random access or nullptr (does c++ initialize pointers?).

Reply:
+1. Added the following snippet:
```
THROW_ERROR_IF_EQUAL(
                   nullptr,
                   service,
                   "Service not found.");
```